clustermesh-apiserver/users_mgmt_test.go

```diff
@@ -113,9 +113,15 @@ func TestUsersManagement(t *testing.T) {
 	require.Equal(t, "r2", client.created["bar"])
 	require.Equal(t, "r3", client.created["qux"])
 
-	// Update the users config file, and require that changes are propagated
 	client.init()
-	require.NoError(t, os.WriteFile(cfgPath, []byte(users2), 0600))
+
+	// Update the users config file, and require that changes are propagated
+	// We first write to a different file and then rename it, to avoid the possible
+	// race condition caused by truncate + write if we detect the event sufficiently
+	// fast (i.e., we first read an empty file, and then the expected one).
+	cfgPath2 := path.Join(tmpdir, "users.yaml.2")
+	require.NoError(t, os.WriteFile(cfgPath2, []byte(users2), 0600))
+	require.NoError(t, os.Rename(cfgPath2, cfgPath))
```

Comment:
Alternatively, this could use `github.com/google/renameio/v2` which would make it clear (without the comment) that the file is written atomically and would also take care of automatically cleaning up stale files on error:

```suggestion
	require.NoError(t, renameio.WriteFile(cfgPath, []byte(users2), 0600))
```

Given that `os.Rename` is an atomic operation on Linux at least, there's no functional difference of the proposed implementation `renameio` implementation (apart from potentially leaking temporary files in case of an error).

Reply:
Thanks! I didn't know there was already a package for that. As for the file leak there should be no problem, since the temporary directory is entirely deleted at the end of the test (deferred after creation).